<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ParameterWindow</class>
 <widget class="QMainWindow" name="ParameterWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>700</width>
    <height>650</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>700</width>
    <height>650</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>700</width>
    <height>650</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Mega Machine Evolution - Parameters</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>4</y>
      <width>351</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Genetic algorithm parameters</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>43</y>
      <width>191</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>Selection mode</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>73</y>
      <width>191</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>Cross-over mode</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>104</y>
      <width>191</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>Mutation mode</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>133</y>
      <width>191</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>Cross-over rate</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>195</y>
      <width>191</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>Selective pressure rate</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>165</y>
      <width>191</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>Mutation rate</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>224</y>
      <width>191</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>Tolerance rate</string>
    </property>
   </widget>
   <widget class="QComboBox" name="cmbSelectionMode">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>40</y>
      <width>461</width>
      <height>26</height>
     </rect>
    </property>
   </widget>
   <widget class="QComboBox" name="cmbCrossOverMode">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>70</y>
      <width>461</width>
      <height>26</height>
     </rect>
    </property>
   </widget>
   <widget class="QComboBox" name="cmbMutationMode">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>100</y>
      <width>461</width>
      <height>26</height>
     </rect>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="dsbCrossOverRate">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>130</y>
      <width>121</width>
      <height>27</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.100000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="dsbMutationRate">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>160</y>
      <width>121</width>
      <height>27</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.100000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="dsbSpRate">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>190</y>
      <width>121</width>
      <height>27</height>
     </rect>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="dsbToleranceRate">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>220</y>
      <width>121</width>
      <height>27</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>359</y>
      <width>351</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>State machine parameters</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>399</y>
      <width>191</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>State numbers</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="sbStateNumbers">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>394</y>
      <width>121</width>
      <height>27</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="value">
     <number>4</number>
    </property>
   </widget>
   <widget class="QPushButton" name="btnRun">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>469</y>
      <width>661</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Run Simulation</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="sbGenerationNumber">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>250</y>
      <width>121</width>
      <height>27</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>1000</number>
    </property>
    <property name="value">
     <number>100</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>254</y>
      <width>191</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>Number of generation</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>285</y>
      <width>191</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>Population size</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="sbPopulationSize">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>281</y>
      <width>121</width>
      <height>27</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>1000</number>
    </property>
    <property name="value">
     <number>100</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>429</y>
      <width>191</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>Max alert</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="sbMaxAlert">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>424</y>
      <width>121</width>
      <height>27</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>1000</number>
    </property>
    <property name="value">
     <number>1</number>
    </property>
   </widget>
   <widget class="QProgressBar" name="pgbGeneration">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>470</y>
      <width>661</width>
      <height>23</height>
     </rect>
    </property>
    <property name="value">
     <number>0</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_14">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>194</y>
      <width>331</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>Enable only for RSP selection mode</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="sbElitePopulationSize">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>310</y>
      <width>121</width>
      <height>27</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>1000</number>
    </property>
    <property name="value">
     <number>5</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_15">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>314</y>
      <width>191</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>Elite population size</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="cbLogMachines">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>390</y>
      <width>181</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Logging in StateMachines</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_16">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>420</y>
      <width>221</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>WARNING : This may take more time and generate big log file</string>
    </property>
    <property name="scaledContents">
     <bool>false</bool>
    </property>
    <property name="wordWrap">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QGroupBox" name="LiveState">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>560</y>
      <width>661</width>
      <height>61</height>
     </rect>
    </property>
    <property name="title">
     <string>Processing</string>
    </property>
    <widget class="QLabel" name="CurrentGen">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>111</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>TextLabel</string>
     </property>
    </widget>
    <widget class="QLabel" name="MaxFitness">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>30</y>
       <width>171</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>TextLabel</string>
     </property>
    </widget>
    <widget class="QLabel" name="MeanFitness">
     <property name="geometry">
      <rect>
       <x>330</x>
       <y>30</y>
       <width>161</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>TextLabel</string>
     </property>
    </widget>
    <widget class="QLabel" name="CurrentAnalysis">
     <property name="geometry">
      <rect>
       <x>530</x>
       <y>30</y>
       <width>121</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>TextLabel</string>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="lblLogFile">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>500</y>
      <width>471</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Current log file location :  log.txt</string>
    </property>
    <property name="wordWrap">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="browseLog">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>510</y>
      <width>171</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Change log file location</string>
    </property>
   </widget>
  </widget>
  <action name="actionLog_file_location">
   <property name="text">
    <string>Log file location</string>
   </property>
   <property name="toolTip">
    <string>Choose log file location</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
